apps/hash-graph/lib/graph/src/api/rest/json_schemas/entity_type.json

```diff
@@ -10,63 +10,150 @@
       ]
     },
     "kind": {
+      "type": "string",
       "enum": ["entityType"]
     },
     "$id": {
-      "type": "string",
-      "format": "uri"
+      "$ref": "#/definitions/VersionedUrl"
     },
     "type": {
+      "type": "string",
       "enum": ["object"]
     },
-    "title": { "type": "string" },
-    "description": { "type": "string" },
+    "title": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    },
+    "allOf": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/EntityTypeReference"
+      }
+    },
     "examples": {
-      "$comment": "Example Entity instances",
       "type": "array",
       "items": {
-        "type": "object",
-        "propertyNames": {
-          "$comment": "Property names must be a valid URL to a Property Type",
-          "type": "string",
-          "format": "uri"
-        }
+        "type": "object"
       }
     },
-    "properties": { "$ref": "#/$defs/propertyTypeObject" },
+    "properties": {
+      "$ref": "#/definitions/PropertyTypeObject"
+    },
     "required": {
       "type": "array",
       "items": {
-        "type": "string",
-        "format": "uri"
+        "$ref": "#/definitions/BaseUrl"
       }
     },
-    "links": { "$ref": "#/$defs/linkTypeObject" }
+    "links": {
+      "type": "object",
+      "propertyNames": {
+        "$ref": "#/definitions/VersionedUrl"
+      },
+      "patternProperties": {
+        "^.*$": {
```

Comment:
This is different from the upstream spec.

Reply:
My apologies, I was also thinking if simply removing them (as you can see the generator doesn't support them.)
`patternProperties` and `propertyNames` are not recognized by OpenAPI 3.0, there is `x-patternProperties` as an unofficial substitute, but I wasn't able to ascertain if OpenAPI-generator supports it (with our current configuration certainly not)

Reply:
So include `patternProperties` (or in this case, `x-patternProperties`)?

Reply:
If `x-patternProperties` is part of the spec let's use that but we should keep the information I think.

Reply:
fixed in [bf1dcfc](https://github.com/hashintel/hash/pull/2838/commits/bf1dcfcdb3ff3a99dad62943e7deb8df9fdf8adf)

(`patternProperties` and `propertyNames` are not supported by `OpenAPI 3.0`, but there are tools that understand the `x-` prefix versions, `x-` is the namespace for any property that isn't strictly OpenAPI 3.0 compliant, also known as extensions)